src/MUIDataTable.js

```diff
@@ -1064,7 +1064,7 @@ class MUIDataTable extends React.Component {
       searchText,
     } = this.state;
 
-    const rowCount = this.options.count || displayData.length;
+    const rowCount = this.props.options.count || this.options.count || displayData.length;
```

Comment:
This is a great catch, and certainly seems to work on my end as well!

However, there might be an opportunity here for an extra fix. There are a lot of table values that currently rely on props and watching for props to change, to refresh the state, however, I think this is a react anti-pattern. I believe it would be better to rely more on state instead, and I think the library should move in this direction.

What that would mean is that we would need to initialize a `count` in the initial state, add `count: options.count,` to this https://github.com/gregnb/mui-datatables/blob/c1cc06fc96c11a1af7ac23b9b856a2c31dca604e/src/MUIDataTable.js#L482 and then change your line to look for `this.state.count` as well.

What do you think? I'm not married to this approach, and I don't want to hold up the fix too long, so we can always go with this and make additional changes later.

Reply:
In my case, the record count and page total will be changed after page changed. So I need a way to refresh it. Currently, I am using solution https://github.com/gregnb/mui-datatables/issues/610#issuecomment-491197237.  Is it the right way to fix this? Or any best ways to meet my case?

Reply:
The solution I suggested above also works, but it gets us away from props, and uses state internally, instead.

Reply:
Good solution, I had used internal `state.count` to instead of `props.count` and it worked.